packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -170,13 +170,14 @@ class RunCommand extends RunCommandBase {
 
     if (argResults['machine']) {
       Daemon daemon = new Daemon(stdinCommandStream, stdoutCommandResponse,
-          notifyingLogger: new NotifyingLogger());
+          notifyingLogger: new NotifyingLogger(), logToStdout: true);
       AppInstance app = daemon.appDomain.startApp(
         device, Directory.current.path, targetFile, route,
         getBuildMode(), argResults['start-paused'], hotMode);
       int result = await app.runner.waitForAppToFinish();
       if (result != 0)
         throwToolExit(null, exitCode: result);
+      return null;
```

Comment:
Good fix. Sorry I missed that [in my earlier PR](https://github.com/flutter/flutter/commit/223f4161996ecd8e950b9375d52a79f3582f69a3#diff-09c5be1877aa83342be68006ff74cb82R167).